mm2src/coins/utxo/qrc20.rs

```diff
@@ -0,0 +1,690 @@
+
+    let (unsigned, data) = try_s!(coin.generate_transaction(unspents, outputs, fee_policy, actual_tx_fee, gas_fee).await);
+    let prev_script = Builder::build_p2pkh(&coin.arc().my_address.hash);
+    let signed = try_s!(sign_tx(unsigned, &coin.arc().key_pair, prev_script, coin.arc().signature_version, coin.arc().fork_id));
+    let fee_details = UtxoFeeDetails {
+        amount: utxo_common::big_decimal_from_sat(data.fee_amount as i64, coin.arc().decimals),
+    };
+    let my_address = try_s!(coin.my_address());
+    let to_address = try_s!(coin.display_address(&to_addr));
+    Ok(TransactionDetails {
+        from: vec![my_address],
+        to: vec![to_address],
+        total_amount: utxo_common::big_decimal_from_sat(data.spent_by_me as i64, coin.arc().decimals),
+        spent_by_me: utxo_common::big_decimal_from_sat(data.spent_by_me as i64, coin.arc().decimals),
+        received_by_me: utxo_common::big_decimal_from_sat(data.received_by_me as i64, coin.arc().decimals),
+        my_balance_change: utxo_common::big_decimal_from_sat(data.received_by_me as i64 - data.spent_by_me as i64, coin.arc().decimals),
+        tx_hash: signed.hash().reversed().to_vec().into(),
+        tx_hex: serialize(&signed).into(),
+        fee_details: Some(fee_details.into()),
+        block_height: 0,
+        coin: coin.arc().ticker.clone(),
+        internal_id: vec![].into(),
+        timestamp: now_ms() / 1000,
+    })
+}
+
+/// Serialize the `number` similar to BigEndian but in QRC20 specific format.
+fn contract_encode_number(number: i64) -> Vec<u8> {
+    // | encoded number (0 - 8 bytes) |
+    // therefore the max result vector length is 8
+    let capacity = 8;
+    let mut encoded = Vec::with_capacity(capacity);
+
+    if number == 0 {
+        return Vec::new();
+    }
+
+    let is_negative = number.is_negative();
+    let mut absnum = (number as i128).abs();
+
+    while absnum != 0 {
+        // absnum & 0xFF is first lowest byte
+        encoded.push((absnum & 0xFF) as u8);
+        absnum >>= 8;
+    }
+
+    if (encoded.last().unwrap() & 0x80) != 0 {
+        encoded.push({ if is_negative { 0x80 } else { 0 } });
+    } else if is_negative {
+        *encoded.last_mut().unwrap() |= 0x80;
+    }
+
+    encoded
+}
+
+fn generate_token_transfer_script_pubkey(
+    to_addr: Address,
+    amount: U256,
+    gas_limit: u64,
+    gas_price: u64,
+    token_addr: &[u8],
+) -> Result<Script, String> {
+    // AP
+    // TODO we should recheck whether Qtum allows to set gas price to zero
+    // it is actually valid situation for ETH, such transactions even get mined
+    // sometimes: https://etherscan.io/tx/0x4f719da4e138bd8ab929f4110e84d773b57376b37d1c635d26cd263d65da99cb
+    // https://medium.com/chainsecurity/zero-gas-price-transactions-what-they-do-who-creates-them-and-why-they-might-impact-scalability-aeb6487b8bb0
+    // I suspect that there might be an implementation error in contract_encode_number function in
+    // qtum electrum wallet.
```

Comment:
Looks like QTUM wallets do not allow setting gas price to 0 so we should consider it as not supported. Please remove this TODO.